### PR TITLE
Feature/cldsrv 244/migrate to GitHub actions

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
Migrate CI from Eve to Github Actions:
* multiple-backend-tests
* file-ft-tests
* kmip-ft-tests
* utapi-v2-tests

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+      - name: Build and push cloudserver image
+        uses: docker/build-push-action@v2.7.0
+        with:
+          push: true
+          context: .
+          tags: |
+            ghcr.io/${{ github.repository }}/cloudserver:${{ github.sha }}
+            registry.scality.com/cloudserver-dev/cloudserver:${{ github.sha }}
+          cache-from: type=gha,scope=cloudserver
+          cache-to: type=gha,mode=max,scope=cloudserver
```
Comment (tcarmet) [inline .github/workflows/tests.yaml:154]: Optional, you can use this reusable workflow if you want, does the same as this job. https://github.com/scality/workflows#docker-build

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        yarn run ft_kmip | tee /tmp/artifacts/${{ github.job }}/tests.log
+    - name: Upload logs to artifacts
+      uses: scality/action-artifacts@v3
+      with:
+        method: upload
+        url: https://artifacts.scality.net
+        user: ${{ secrets.ARTIFACTS_USER }}
+        password: ${{ secrets.ARTIFACTS_PASSWORD }}
+        source: /tmp/artifacts
+      if: always()
```
Comment (tcarmet) [inline .github/workflows/tests.yaml]: This is optional as well, and should be done in another PR given the time we have:

I believe you can optimize those jobs with a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs), with something that looks like this:
```yaml
jobs:
  tests:
    strategy:
      matrix:
        tests:
        # more var/settings can be added as needed
        - name: multiple-backend
          env:
            CLOUDSERVER_IMAGE: ghcr.io/${{ github.repository }}/cloudserver:${{ github.sha }}
            S3BACKEND: mem
            S3_LOCATION_FILE: /usr/src/app/tests/locationConfig/locationConfigTests.json
            S3DATA: multiple
            JOB_NAME: ${{ github.job }}
        - name: kmip-ft-tests
        ...
        # add the settings for all tests
    name: ${{ matrix.tests.name }}
    env: ${{ matrix.tests.env }}
    steps:
    # Define the steps a single time and use the ${{ matrix.tests.<var> }} as needed for templating
    
```

**Review by tcarmet — APPROVED** [review]
Overall this LGTM, I have one required change that can be done in a separate PR and some ideas for you to play with in the future.

## Required changes

Stuff that are missing but can be done in another PR
- [ ] You need a release workflow, which seems to be missing here. You can pretty much copy paste the one from your neighboors like [backbeat](https://github.com/scality/backbeat/blob/development/8.6/.github/workflows/release.yaml). 

## Optimization

- [ ] I would split the `linting-coverage` in another workflow, just to keep files smaller and ease up reviews. 
- [ ] It also seems that `linting-coverage` does more than linting, which you may dedicate a job for the unit tests that are executed there.
- [ ] Create a reusable workflow for the end to end tests of cloudserver (might be overkill though).

**Comment by alexanderchan-scality** [comment]
thanks @tcarmet.
The release workflow will be added for CLDSRV-329 in separate PR that is specific to migrating the 8.x branches to actions.

**Comment by tcarmet** [comment]
> thanks @tcarmet.
> The release workflow will be added for [CLDSRV-329](https://scality.atlassian.net/browse/CLDSRV-329) in separate PR that is specific to migrating the 8.x branches to actions.

Alright perfect 👌 

I'll let you double check if its specific to 8.x, as it seems that federation 7.10 is pulling images from cloudserver 7.10: https://github.com/scality/Federation/blob/development/7.10/group_vars/all#L138

UPDATE: I just understood, it already exists as a github actions workflow for 7.10, so indeed the work will be to just adapt the code for 8.x context

[CLDSRV-329]: https://scality.atlassian.net/browse/CLDSRV-329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by nicolas2bert — APPROVED** [review]

**Comment by alexanderchan-scality** [comment]
ping

**Comment by alexanderchan-scality** [comment]
@tcarmet will need some assistance merging this and possibly also with the removal of the eve webhook.

**Comment by alexanderchan-scality** [comment]
ping
